src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -377,38 +378,313 @@ internal PhysicalLocationVersionOne CreatePhysicalLocation(Region v2Region)
+                        // IT MEANS "THE REST OF THE StartLine". BUT IF CHARLENGTH IS
+                        // PRESENT BUT 0, IT MEANS "INSERTION POINT". AND WE CAN'T TELL
+                        // THE DIFFERENCE.
+                        // TODO: Issue #932
+
+                        // Assume it's an insertion point
+                        if (v2Region.StartLine > 0)
+                        {
+                            region.EndLine = region.StartLine;
+                            region.EndColumn = region.StartColumn;
+                        }
+                        else
+                        {
+                            int endLine, endColumn;
+                            GetRegionStart(v2Region, uri, out endLine, out endColumn);
+
+                            region.EndLine = endLine;
+                            region.EndColumn = endColumn;
+                        }
+                    }
+                }
+                else
+                {
+                    // There are no text-related properties. Therefore either the region is
+                    // described entirely by binary-related properties, or the region is an
+                    // insertion point at the start of the file.
+                    region.Length = v2Region.ByteLength;
+                    region.Offset = v2Region.ByteOffset;
+                }
             }
 
             return region;
         }
 
+        private void GetRegionStart(Region v2Region, Uri uri, out int startLine, out int startColumn)
+        {
+            startLine = 0;
+            startColumn = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                // Read charOffset characters
+                char[] buffer = new char[v2Region.CharOffset];
+                reader.Read(buffer, 0, buffer.Length);
+
+                string text = new string(buffer);
+                string[] parts = text.Split(new string[] { "\r\n" }, StringSplitOptions.None);
+                startLine = parts.Length;
+                startColumn = parts.Last().Length + 1;
+            }
+            finally
+            {
+                if (reader != null)
+                {
+                    reader.Dispose();
+                }
+            }
+        }
+
+        private int GetRegionByteOffset(Region v2Region, Uri uri)
+        {
+            int result = 0;
+            TextReader reader = null;
+
+            try
+            {
+                Encoding encoding;
+                reader = GetFileTextReader(uri, out encoding);
+
+                if (reader != null)
+                {
+                    char[] buffer = new char[v2Region.CharOffset];
+
+                    // Read everything up to charOffset
+                    if (reader.ReadBlock(buffer, 0, buffer.Length) > 0)
```

Comment:
Interesting. I didn't know about `ReadBlock` and I had to look up [the difference](https://stackoverflow.com/questions/10497575/what-is-the-difference-between-read-and-readblock-calls-on-textreader) between it and `Read`. I agree, `ReadBlock` is appropriate here.